### DISCUSSION TITLE
Drawing sloped surface

### DISCUSSION THREAD

**Discussion by ahmedahmed2056** [firemodels/fds#13113] [General]
![image](https://github.com/firemodels/fds/assets/165714562/6957dd64-c287-4f8b-9c50-7a7a6eb33c2f)

here for recreating the ground surfaces of this case, i did the following: 
&VENT XB=45.0,57.0,0,36.0,0,0, SURF_ID='ROAD' /
&VENT XB=37.0,45.0,0,36.0,0,8.0, SURF_ID='ROAD' /
&VENT XB=0,37.0,0,36.0,8.0,8.0, SURF_ID='ROAD' /

- for the first line (surface A) it has no problems and successfully done.
- for the second line , the sloped surface (surface b), it gives me the following error 
   ERROR(810): VENT 6 must be a plane.
- for the third line (surface c), it gives the following error 
   ERROR(820): VENT 6 has no solid backing or an orientation index (IOR) is needed.

So any suggestions to model the sloped surface by 30 degree (surface b) and the upper ground on which the building is standing (surface c) ?

**Comment by marcosvanella** [comment]
I suggest you use a preprocessor like Pyrosym or BlenderFDS to do this. This way you can apply VENTs and SURF_IDs in your model and export the geometry to an FDS input file.

**Reply by drjfloyd** [reply]
To expand a little bit.

If you are using GEOM for your model then the User's Guide states in 7.3.11:  A VENT cannot be applied to a GEOM. Surface properties must be applied face by face via SURF_ID.

If you are using OBST, then VENT inputs must be planes. You would need to define a series of VENT inputs to cover the top and side of the OBST in the region of the slope.

**Reply by ahmedahmed2056** [reply]
Thanks for your reply. But still the problem in pyrosim or fds  is how to do the sloped plane ?
I am trying to search for a function for that in the user guide but can't find.

**Reply by marcosvanella** [reply]
You should be able to put a wedge shape with the slope in pyrosym, put in it the SURF_ID you want and export it as voxels (OBSTs) or a single GEOM to the FDS input file.
Another option directly in the input file would be to use an extruded polygon GEOM, look it up in the user guide.

**Reply by Sun0036** [reply]
Hello, actuelly, i work with ahmed for this model, i have a question for the slope, for wedge shape slope, i made it，but what we wanted to achieve is a smooth slope，so can we use this function to constructe a smooth slope? It doesn't run currently
&OBST ID='Slope'
      XB=12.0, 20.0, 0.0, 36.0, 0, 8.0,
      SURF_IDS='CONCRETE_SURF',
      MB_TYPE='PLANE',
      VERTICES=12 0 0 12 36 0 20 36 8.0 20 0 8.0 /

Thank you!

**Reply by marcosvanella** [reply]
Well, MB_TYPE and VERTICES are not fields of the &OBST namelist. 
I suggest you get the latest release of FDS (6.9.1) and read up on sections 7.2 (rectilinear obstructions) and 7.3 (unstructured geometries).

**Reply by Sun0036** [reply]
Yeah, but in the papar, they used FDS 6.7.7 to achieve this slope, is there any proposition for making this slope?

**Reply by marcosvanella** [reply]
My suggestions are what I commented already. If you are using Pyrosim to generate your model, there has to be a way to add a wedge or quadrilateral shape that will have the slope you need. You might even make it in some CAD software and import it. From then on it is a matter of applying boundary conditions (SURF_IDs) on that object and export it to an fds input file, either as a collection of OBSTs or a GEOM. If you have questions on how to go about this you can seek advice in the Pyrosym user forum. 
Also, you want to use the latest version of FDS, it will have whatever options were used in 6.7.7 for this problem, plus many changes and bug fixes done since. Besides, it is the only version we maintain.

**Comment by obscureed** [comment]
I can't find where it says this in the User Guide, but here are some facts about FDS: The mesh blocks are cuboids, aligned with the x/y/z axes; the mesh blocks are divided into cuboid cells; and each cell is either 100% fluid or 100% obstacle. (OK, there are some complications: There can be infinitely-thin obstacles. A cell can switch between fluid and obstacle at different times. There are beta features relating to unstructured geometry, which includes angled surfaces. I'll ignore these for the sake of clarity. The beta features did not exist at the time of FDS version 6.7.7.)  So, if you are trying to copy the geometry of an FDS model in a paper -- you don't mention which paper -- then that geometry presumably had stair-stepped blockwise representations of slopes. There are no commands for angled surfaces, because angled surfaces do not exist (still ignoring the beta features). You might be interested in the "pyramid" demo in the FDS User Guide.
